deeptime/markov/msm/tram/_tram.py

```diff
@@ -0,0 +1,708 @@
+        self.count_models.
+        Based on each TransitionCountModel, construct state_count and transition_count matrices that contain the counts
+        for each thermodynamic state. These are reshaped to contain all possible markov states, even the once without
+        counts. This is done so that the _tram_bindings receives count matrices that are all the same shape, which is
+        easier to handle (matrices are padded with zeros for all empty states that got dropped by the
+        TransitionCountModels).
+
+        Parameters
+        ----------
+        dtraj_fragments: list(List(Int))
+           A list that contains for each thermodynamic state the fragments from all trajectories that were sampled at
+           that thermodynamic state. fragment_indices[k][i] defines the i-th fragment sampled at thermodynamic state k.
+           The fragments should be restricted to the largest connected set and not contain any negative state indices.
+
+        Returns
+        -------
+        transition_counts: ndarray(n, m, m)
+            The transition counts matrices. transition_counts[k] contains the transition counts for thermodynamic state
+            k, restricted to the largest connected set of state k.
+        state_counts: ndarray(n, m)
+            The state counts histogram. state_counts[k] contains the state histogram for thermodynamic state k,
+            restricted to the largest connected set of state k.
+        """
+
+        estimator = TransitionCountEstimator(lagtime=self.lagtime, count_mode=self.count_mode)
+
+        transition_counts = np.zeros((self.n_therm_states, self.n_markov_states, self.n_markov_states), dtype=np.intc)
+        state_counts = np.zeros((self.n_therm_states, self.n_markov_states), dtype=np.intc)
+
+        self.count_models = []
+        for k in range(self.n_therm_states):
+
+            if len(dtraj_fragments[k]) == 0:
+                # there are no samples from this state that belong to the connected set. Make an empty count model.
+                self.count_models.append(TransitionCountModel(np.zeros(self.n_markov_states, self.n_markov_states)))
+            else:
+                # make a counts model for the samples that belong to the connected set.
+                traj_counts_model = estimator.fit_fetch(dtraj_fragments[k])
+                # create index for all elements in transition matrix
+                i_s, j_s = np.meshgrid(traj_counts_model.state_symbols, traj_counts_model.state_symbols)
+
+                # place submodel counts in our full-sized count matrices
+                transition_counts[k, i_s, j_s] = traj_counts_model.count_matrix.T
+                state_counts[k, traj_counts_model.state_symbols] = traj_counts_model.state_histogram
+
+                self.count_models.append(traj_counts_model)
+        return state_counts, transition_counts
+
+    def _to_markov_model(self):
```

Comment:
perhaps better `_construct_markov_model` and return the MSM collection to set the `_model` inside the main `fit` method? that way there is less indirection and coupling between methods